dev-envs/linux/tools.sh

```diff
@@ -181,10 +181,10 @@ go install github.com/go-delve/delve/cmd/dlv@latest
 go install github.com/josharian/impl@latest
 go install github.com/fatih/gomodifytags@latest
 
-GOLANGCI_LINT_VERSION="$(curl "${curl_opts[@]}" https://raw.githubusercontent.com/DataDog/datadog-agent/main/internal/tools/go.mod | awk -Fv '/golangci-lint/ {print $2}')"
+GOLANGCI_LINT_VERSION="$(curl "${curl_opts[@]}" https://raw.githubusercontent.com/DataDog/datadog-agent/7.73.x/internal/tools/go.mod | awk -Fv '/golangci-lint/ {print $2}')"
```

Comment:
on main, datadog-agent is using golangci-lint v2.7.0 and this is causing a malformed url when trying to download - we do not want to upgrade the golangci-lint version in a backport so lets pin this to 7.73.x for the backport

Reply:
SIgh....  I feel your pain. 
The lint version should be pinned to the agent code, not dda.